### PR TITLE
HPCC-12846 Std.System.Util.CmdProcess() did not process arguments to command

### PR DESCRIPTION
**Pull request by richardkchapman**
Replaced jlib function that was designed just to pass environment (and was
only called from otherwise unused code), with one that handles parameters
(moved from ecl-bundle.cpp).

Signed-off-by: Richard Chapman rchapman@hpccsystems.com

### CODE REVIEW

**Comment by richardkchapman** [comment]
@afishbeck Please review

**Comment by afishbeck** [comment]
@richardkchapman looks good

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: system/jlib/jutil.cpp [hunk 1]
```diff
@@ -1887,6 +1712,44 @@ static const char *findExtension(const char *fn)
...
+    {
+        Owned<IPipeProcess> pipe = createPipeProcess();
+        pipe->run(cmd, cmd, ".", input != NULL, true, true, 1024*1024);
+        if (input)
+        {
+            pipe->write(strlen(input), input);
+            pipe->closeInput();
+        }
+        char buf[1024];
+        while (true)
```
Comment (ghalliday) [inline system/jlib/jutil.cpp]: Does this need multiple threads - e.g., if so much error output is generated that output to stderr blocks so no output ever goes to stdout?  If the input can be large can you have similar problems?

Comment (richardkchapman) [inline system/jlib/jutil.cpp]: You can. But that would be a separate bug from the one that this change is fixing.

**Comment by ghalliday** [comment]
Added https://track.hpccsystems.com/browse/HPCC-12868

**Pull request merged**